clients/payment/src/components/CardPayment.tsx

```diff
@@ -104,52 +106,73 @@ export const CardPayment = ({ paymentLink, paymentMethodId, publicKey }: Props)
         expiryYearPlaceholder: "YY",
       },
       schemeChoice: true,
+      acceptedPaymentMethods: ["Visa", "Maestro", "Mastercard", "Cartes Bancaires"],
     });
+  }, [publicKey]);
+
+  useEffect(() => {
+    Frames.addEventHandler(
+      "paymentMethodChanged",
+      //@ts-expect-error addEventHandler isn't typed correctly
+      (event: { paymentMethod: string; isPaymentMethodAccepted: boolean; isValid: boolean }) => {
```

Comment:
Instead of using `ts-expect-error`, you could validate the event shape using `isMatching` from `ts-pattern` (safer)